### PR TITLE
Add rule example

### PR DESCRIPTION
**Pull request by DevHaoZi**

### CODE REVIEW

File: app/rules/exists.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	if err != nil {
+		return false
+	}
+
+	return count != 0
+}
+
+// Message Get the validation error message.
+func (receiver *Exists) Message() string {
+	return "记录不存在"
```
Comment (hwbrzzl) [inline app/rules/exists.go:54]:
```suggestion
	return "record does not exist"
```

File: app/rules/not_exists.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	if err != nil {
+		return false
+	}
+
+	return count == 0
+}
+
+// Message Get the validation error message.
+func (receiver *NotExists) Message() string {
+	return "记录已存在"
```
Comment (hwbrzzl) [inline app/rules/not_exists.go:54]:
```suggestion
	return "record already exists"
```

File: app/rules/not_exists.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+// Passes Determine if the validation rule passes.
+func (receiver *NotExists) Passes(_ validation.Data, val any, options ...any) bool {
+
+	tableName := options[0].(string)
+	fieldName := options[1].(string)
+	requestValue := val.(string)
+
+	// Check if it is empty
```
Comment (hwbrzzl) [inline app/rules/not_exists.go:32]:
```suggestion
```

File: app/rules/exists.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+// Passes Determine if the validation rule passes.
+func (receiver *Exists) Passes(_ validation.Data, val any, options ...any) bool {
+
+	tableName := options[0].(string)
+	fieldName := options[1].(string)
+	requestValue := val.(string)
+
+	// Check if it is empty
```
Comment (hwbrzzl) [inline app/rules/exists.go:32]:
```suggestion
```

**Review by hwbrzzl — COMMENTED** [review]
Perfect, just a few nitpicks.

**Comment by DevHaoZi** [comment]
fixed

**Review by hwbrzzl — APPROVED** [review]
LGTM

**Pull request merged**